Review.swift

```diff
@@ -0,0 +1,294 @@
+            logger.info("\(error.localizedDescription)")
+        }
+    }
+    
+    func directoryExists(pullRequest: PullRequest) -> Bool {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        let directoryExists = fileManager.fileExists(atPath: path + "sources_" + shortHash)
+        logger.info("Directory exists: \(directoryExists.description)")
+        
+        return directoryExists
+    }
+    
+    func createDirectory(pullRequest: PullRequest) {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        do {
+            logger.info("Creating directory: sources_\(shortHash)")
+            try fileManager.createDirectory(atPath: path + "sources_" + shortHash, withIntermediateDirectories: false, attributes: nil)
+        } catch {
+            logger.info("\(error.localizedDescription)")
+        }
+    }
+    
+    func unzipSources(fileName: String, fileExtension: String, pullRequest: PullRequest) {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        do {
+            logger.info("Unzip sources...")
+            try fileManager.unzipItem(at: URL(fileURLWithPath: path + "sources_" + shortHash + "/" + fileName + fileExtension), to: URL(fileURLWithPath: path + "sources_" + shortHash))
+        } catch {
+            logger.info("\(error.localizedDescription)")
+        }
+    }
+    
+    func generateTasks(for violations: [StyleViolation]) -> [Task] {
+        let ruleDescriptionNames = violations.map { violation in
+            violation.ruleName
+        }
+        
+        let repeatedElements = repeatElement(1, count: ruleDescriptionNames.count)
+        
+        let tasksDictionary = Dictionary(zip(ruleDescriptionNames, repeatedElements), uniquingKeysWith: +)
+        
+        return tasksDictionary.map { task in
+            return Task(description: task.key, occurence: task.value)
+        }
+    }
+    
+    func generateResult(for violations: [StyleViolation]) -> [String:Int] {
+        let ruleDescriptionNames = violations.map { violation in
+            violation.ruleName
+        }
+        
+        let repeatedElements = repeatElement(1, count: ruleDescriptionNames.count)
+        
+        let resultDictionary = Dictionary(zip(ruleDescriptionNames, repeatedElements), uniquingKeysWith: +)
+        
+        return resultDictionary
+    }
+    
+    func getCommentMetaData(violations: [StyleViolation], diff: Diff, onlyAddedLines: Bool = false) -> ([StyleViolation], [(StyleViolation, Segment)]) {
```

Comment:
🚫 Error
 [Line Length] At line 254:
Line should be 150 characters or less: currently 152 characters